File: examples/php/VideoService.php
<?php
// Interface: Definiert die gemeinsame Sprache für Proxy und echtes Objekt
interface IVideoService {
    public function playVideo(string $id): void;
}

// Echtes Objekt: Die ressourcenintensive Klasse
class YouTubeService implements IVideoService {
    public function playVideo(string $id): void {
        echo "Streaming Video: " . $id . "...\n";
    }
}

File: examples/csharp/VideoService.cs
using System;

namespace ProxyPatternExamples 
{
    // Interface
    public interface IVideoService {
        void PlayVideo(string id);
    }

    // Echtes Objekt
    public class YouTubeService : IVideoService {
        public void PlayVideo(string id) {
            Console.WriteLine($"Streaming Video: {id}...");
        }
    }
}
